kornia/augmentation/augmentation.py

```diff
@@ -21,45 +21,45 @@
 
 
 class AugmentationBase(nn.Module):
-    def __init__(self, apply_fcn: Callable, return_transform: bool = False) -> None:
+    def __init__(self, return_transform: bool = False) -> None:
         super(AugmentationBase, self).__init__()
         self.return_transform = return_transform
-        self._apply_fcn: Callable = apply_fcn
-        self._params: Dict[str, torch.Tensor] = {}
 
     def infer_batch_shape(self, input: UnionType) -> torch.Size:
         return _infer_batch_shape(input)
 
-    def get_params(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
+    def generate_parameters(self, input_shape: torch.Size) -> Dict[str, torch.Tensor]:
         raise NotImplementedError
 
-    def forward(self, input: UnionType, params: Optional[Dict[str, torch.Tensor]] = None) -> UnionType:  # type: ignore
+    def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
+        raise NotImplementedError
+
+    def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
```

Comment:
This naming it's not generic. Since we can have here non geometric augmentations

Reply:
```apply``` is used in ```nn.Module``` internally. We cannot override it. So I used apply_transform here. Let me know if you got better naming.

Reply:
oki